### PR TITLE
Display DataRef as autocomplete changes

### PR DESCRIPTION
**Pull request by 4manasa**

### CODE REVIEW

File: src/components/forms/AutoComplete/index.tsx [hunk 1]
```diff
@@ -1,74 +1,146 @@
...
 
+declare const PCore;
 
 interface IOption {
   key: string;
   value: string;
 }
 
-export default function AutoComplete(props) {
+export const getDisplayFieldsMetaData = columns => {
```
Comment (JEAxero) [inline src/components/forms/AutoComplete/index.tsx:15]: I'm not sure why this function is outside the Autocomplete function scope. It doesn't seem to be used by other components. Should it be inside Autocomplete?

File: src/components/forms/AutoComplete/index.tsx [hunk 2]
```diff
@@ -1,74 +1,146 @@
...
+    if (displayColumns[index].primary === 'true') {
+      metaDataObj.primary = displayColumns[index].value;
+    } else {
+      metaDataObj.secondary.push(displayColumns[index].value);
+    }
+  }
+  return metaDataObj;
+};
+
+export const preProcessColumns = columns => {
```
Comment (JEAxero) [inline src/components/forms/AutoComplete/index.tsx:30]: I'm not sure why this function is outside the Autocomplete function scope. It doesn't seem to be used by other components. Should it be inside Autocomplete?

**Review by JEAxero — COMMENTED** [review]
See questions about whether we can move the 2 helper functions inside the component function.

File: src/components/forms/AutoComplete/index.tsx (hunk 1 above)
Comment (4manasa) [reply src/components/forms/AutoComplete/index.tsx]: moved the above methods inside Autocomplete function.

**Review by JEAxero — APPROVED** [review]
Tested with sample app and looks good to me. Thanks for the updates.

**Pull request merged**